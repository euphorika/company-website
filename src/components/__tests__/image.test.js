import React from "react"
import renderer from "react-test-renderer"

import Image from "../image"

describe("Image", () => {
  it("renders correctly", () => {
    const img = {
      aspectRatio: 1.5,
      src: 'image1.jpg',
      srcSet: 'set1, set2',
      sizes: '100vw'
    }

    const tree = renderer
      .create(<Image fluid={img} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
