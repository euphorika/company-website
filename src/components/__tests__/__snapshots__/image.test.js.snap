// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image renders correctly 1`] = `
<div
  className="imageContainer"
>
  <div
    className=" gatsby-image-wrapper"
    style={
      Object {
        "height": "100%",
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <div
      style={
        Object {
          "paddingBottom": "66.66666666666667%",
          "width": "100%",
        }
      }
    />
    <picture>
      <img
        alt=""
        onLoad={[Function]}
        sizes="100vw"
        src="image1.jpg"
        srcSet="set1, set2"
        style={
          Object {
            "fontFamily": "\\"object-fit: cover; object-position: 30% 50%\\"",
            "height": "100%",
            "left": 0,
            "objectFit": "cover",
            "objectPosition": "30% 50%",
            "opacity": 0,
            "position": "absolute",
            "top": 0,
            "transition": "opacity 0.5s",
            "width": "100%",
          }
        }
      />
    </picture>
    <noscript
      dangerouslySetInnerHTML={
        Object {
          "__html": "<picture><img sizes=\\"100vw\\" srcset=\\"set1, set2\\" src=\\"image1.jpg\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
        }
      }
    />
  </div>
</div>
`;
